pipenv/environments.py

```diff
@@ -299,12 +299,16 @@ def initialize(self):
 
         self.PIPENV_VENV_IN_PROJECT = os.environ.get("PIPENV_VENV_IN_PROJECT")
         if self.PIPENV_VENV_IN_PROJECT is not None:
-            self.PIPENV_VENV_IN_PROJECT = (
-                self.PIPENV_VENV_IN_PROJECT.lower() in _true_values
-            )
-        """If set, creates ``.venv`` in your project directory.
-
-        Default is to create new virtual environments in a global location.
+            if self.PIPENV_VENV_IN_PROJECT.lower() in _true_values:
+                self.PIPENV_VENV_IN_PROJECT = True
+            elif self.PIPENV_VENV_IN_PROJECT.lower() in _false_values:
+                self.PIPENV_VENV_IN_PROJECT = False
```

Comment:
Why does one need both `False` and `None` here? I see you documented it, but why do we need this feature?
If this is only for backward compatibility, I suggest considering supporting it but also to issue a warning that only `false`
should be used in the future.
If someone needs `.venv` they could specify it specifically. Otherwise, IMHO `.venv` is like magic. It's hidden at first when you ` ls` a directory, and this behavior is kind of opposite to Python's own: Explicit it better than implicit.
This is a great principle for any software project too...

Reply:
None indicates that you haven't set the variable, so in this case the behavior is use the .venv if it is present.  However if you want to definitely ignore a present .venv you would se it to be False.   This is helpful for those doing development locally and building docker containers, that sometimes the .venv directory exists locally and gets copied into their container but is different system installed versions than the OS the container is running on.  They want to be able to explicitly state in the .env not to ever use a .venv directory in the project, even if it exists because it got copied into the container.  For additional context:  https://github.com/pypa/pipenv/issues/2763#issuecomment-1083715772

Reply:
> if you want to definitely ignore a present .venv you would se it to be False.

I thought this should be the default, always ignore. Even if the variable isn't set.

A solution to @smac89's problem could done with `.dockerignore`. They probably have something like this:
in project source:
```
$ ls -a
setup.py
Pipfile
Pipfile.lock
src/<many-python-files-here>
.venv
Dockerfile
```

Now, they (naively) do something in `Dockerfile`
```
ADD  .  /usr/src/app
WORKDIR  /usr/src/app
RUN pipenv install
```
And viola, this thing is broken, because it uses an existing `venv`. The solution is to create `.dockerignore` file, and then add `.venv` in it. This will prevent the ADD directive from copying `.venv` into the docker container.

The real problem is however, the decision to [default  to the name](https://github.com/pypa/pipenv/blob/9a3b3ce70621af6f9adaa9eeac9cf83fa149319c/pipenv/environments.py#L301) `.venv`. This should have been named either `pipenv` `venv` without the leading dot.

Reply:
As I mentioned in my comment, we use `docker-compose`, and as far as I can tell, docker-compose does not give one the option of filtering out certain files from a mounted volume. Therefore, once we mount the volume inside the build container, it includes everything in that volume.